src/main/java/com/coreos/jetcd/EtcdClient.java

```diff
@@ -14,15 +23,40 @@
     private final ManagedChannel channel;
 
     private final EtcdKV kvClient;
+    private final EtcdAuth authClient;
+
+    private KVGrpc.KVFutureStub kvStub;
+    private AuthGrpc.AuthFutureStub authStub;
 
     public EtcdClient(ManagedChannelBuilder<?> channelBuilder, EtcdClientBuilder builder) {
-        this.endpoints = (String[])builder.endpoints().toArray();
-        this.channelBuilder = channelBuilder != null ? channelBuilder : ManagedChannelBuilder.forAddress("localhost", 2379);
+        this.endpoints = new String[builder.endpoints().size()];
+        builder.endpoints().toArray(this.endpoints);
+        this.channelBuilder = channelBuilder != null ? channelBuilder : ManagedChannelBuilder.forAddress("localhost", 2379).usePlaintext(true);
+
         this.channel = this.channelBuilder.build();
 
-        KVGrpc.KVFutureStub kvStub = KVGrpc.newFutureStub(this.channel);
+        kvStub = KVGrpc.newFutureStub(this.channel);
+        authStub = AuthGrpc.newFutureStub(this.channel);
+
+        String token=null;
+
+        if(builder.getName()!=null && builder.getPassword()!=null){
+            token = getToken(this.channel, builder.getName(), builder.getPassword());
+        }
+
+        if(token!=null) {
+            /**
+             * add token to channel header
+             */
+            Metadata metadata = new Metadata();
+            Metadata.Key<String> TOKEN_KEY= Metadata.Key.of("token", Metadata.ASCII_STRING_MARSHALLER);
```

Comment:
let's use withCallCredential. That is more "official" way to do this sort of thing